### ISSUE TITLE
Management templates from azurerm_key_vault_access_policy data source return values in lowercase

### ISSUE THREAD

**Issue by abergie** [hashicorp/terraform-provider-azurerm#17086]
### Is there an existing issue for this?

- [X] I have searched the existing issues

### Community Note

* Please vote on this issue by adding a :thumbsup: [reaction](https://blog.github.com/2016-03-10-add-reactions-to-pull-requests-issues-and-comments/) to the original issue to help the community and maintainers prioritize this request
* Please do not leave "+1" or "me too" comments, they generate extra noise for issue followers and do not help prioritize the request
* If you are interested in working on this issue or have submitted a pull request, please leave a comment

### Terraform Version

1.2.2

### AzureRM Provider Version

3.8.0

### Affected Resource(s)/Data Source(s)

azurerm_key_vault_access_policy

### Terraform Configuration Files

```hcl
data "azurerm_key_vault_access_policy" "admin_access_policy" {
  name = "Key, Secret, & Certificate Management"
}
```

### Debug Output/Panic Output

```shell
Error: expected access_policy.0.key_permissions.0 to be one of [Backup Create Decrypt Delete Encrypt Get Import List Purge Recover Restore Sign UnwrapKey Update Verify WrapKey], got get
```

### Expected Behaviour

Data source management templates should return the values in correct format (TitleCase)

### Actual Behaviour

Values are lowercase, and are being provided from here: https://github.com/Azure/azure-sdk-for-go/blob/main/services/keyvault/mgmt/2021-10-01/keyvault/enums.go

### References

#16354

**Comment by sinbai** [comment]
@abergie thanks for opening this issue here. The fix has been released in [v3.19.0 of the Terraform Provider](https://github.com/hashicorp/terraform-provider-azurerm/blob/v3.19.0/CHANGELOG.md). However, it's worth mentioning that the fix would be available after v4.0 of terraform provider as this is a breaking change. Could you keep track of it and see if it works for you?

**Comment by mybayern1974** [comment]
Per sinbai's suggestion, I'm closing this issue. Feel free to call out this issue needs to be re-activated.
